scripts/apps/authoring/authoring/controllers/AssociationController.ts

```diff
@@ -159,11 +159,14 @@ export function AssociationController(config, send, api, $q, superdesk,
      * @param {Function} callback to call after save
      */
     this.updateItemAssociation = function(scope, updated, customRel, callback = null, autosave = false) {
-        let data = {}, rel = customRel || scope.rel;
+        let data = {};
+        // if the media is of type media-gallery, update same association-key not the next one
+        // as the scope.rel contains the next association-key of the new item
```

Comment:
> update same association-key not the next one as the scope.rel contains the next association-key of the new item

Why there are multiple keys and multiple items? Could you explain how this works?

Reply:
A field of type media-gallery can hold multiple items, so here https://github.com/superdesk/superdesk-client-core/blob/master/scripts/apps/authoring/authoring/directives/ItemCarouselDirective.ts#L62  the `scope.rel` contains the next key where a media item can be added. Like if I have added one media with 'media--1` key then `scope.rel` will hold the next key as `media--2`.